components/renku_data_services/resource_pool_adapters/adapter.py

```diff
@@ -509,6 +523,12 @@ async def update_user_resource_pools(
                     raise errors.MissingResourceError(
                         message=f"The resource pools with ids: {missing_rps} do not exist."
                     )
+                if user.no_default_access:
+                    default_rp = next((rp for rp in rps_to_add if rp.default), None)
+                    if default_rp:
+                        raise errors.NoDefaultPoolAccessError(
+                            message=f"User with keycloak id {keycloak_id} cannot access the default resource pool"
```

Comment:
I did this but now the error message is misleading since it says that the pool doesn't exist while it can exist but be the default pool which the user cannot access.
If we want to distinguish between these two cases, then we need to execute the postgre query two times which is more costly than filtering in python.
Let's discuss this tomorrow.

Reply:
We'll modify the error message to reflect both cases.